pkg/virtual/workspaces/registry/rest.go

```diff
@@ -163,27 +150,29 @@ func (s *REST) NamespaceScoped() bool {
 	return false
 }
 
-func (s *REST) getPrettyNameFromInternalName(user kuser.Info, internalName string) (string, error) {
+func (s *REST) getPrettyNameFromInternalName(user kuser.Info, orgName, internalName string) (string, error) {
 	list, err := s.crbInformer.Informer().GetIndexer().ByIndex(InternalNameIndex, internalName)
 	if err != nil {
 		return "", err
 	}
 	for _, el := range list {
 		if crb, isCRB := el.(*rbacv1.ClusterRoleBinding); isCRB &&
+			crb.ClusterName == orgName &&
 			len(crb.Subjects) == 1 && crb.Subjects[0].Name == user.GetName() {
 			return crb.Labels[PrettyNameLabel], nil
 		}
 	}
 	return "", kerrors.NewNotFound(tenancyv1beta1.SchemeGroupVersion.WithResource("workspaces").GroupResource(), internalName)
 }
 
-func (s *REST) getInternalNameFromPrettyName(user kuser.Info, prettyName string) (string, error) {
+func (s *REST) getInternalNameFromPrettyName(user kuser.Info, orgName, prettyName string) (string, error) {
```

Comment:
orgClusterName

Reply:
Fixed in commit c48becc80ef541475a0e032a26cfa6a9024c5ac8